### PR TITLE
Add support for iOS, watchOS and tvOS

### PR DESCRIPTION
**Pull request by conradev**
This adds support for all known Apple platforms

I was upgrading my iOS/macOS/Android [app](https://github.com/conradev/Wallet) to Kotlin 1.9.10 in the process of replacing my usage of `kotlinx-cli` with `clikt`, I realized that it doesn't have support for iOS. I added support to `clikt`, and it depends on `mordant` and thus here I am 😄 

This approach works well for [kotlin-logging](https://github.com/oshai/kotlin-logging/blob/a7caeef24ecbd392907d377a3120f4515d4ac5a9/build.gradle.kts#L71-L83), which I use for logging in my app.

I use command line arguments on iOS to configure the binary for integration testing:

Screenshot 2023-09-09 at 7 35 27 PM

### CODE REVIEW

File: mordant/build.gradle.kts [hunk 1]
```diff
@@ -44,9 +55,10 @@ kotlin {
         val mingwX64Main by getting { dependsOn(nativeMain) }
         val posixMain by creating { dependsOn(nativeMain) }
         val linuxX64Main by getting { dependsOn(posixMain) }
-        val macosMain by creating { dependsOn(posixMain) }
-        listOf("macosX64", "macosArm64").forEach { target ->
-            getByName(target + "Main").dependsOn(macosMain)
+        val darwinMain by creating { dependsOn(posixMain) }
+        darwinTargets.forEach {
+            getByName("${it.targetName}Main").dependsOn(darwinMain)
+            getByName("${it.targetName}Test").dependsOn(commonTest)
```
Comment (ajalt) [inline mordant/build.gradle.kts:61]: Is `dependsOn(commonTest)` necessary? I thought commonMain and commonTest were implicit.

**Comment by ajalt** [comment]
Thanks for the PR, and thanks for the use case example; I would have wondered how a CLI could be used on an iphone.

Let add these targets to the github actions tests for the macos runner (or at least all of the targets that can be run from ios).

Do ios, watchos, and tvos have identical syscalls to macos? Otherwise we'll need to write new versions of `MppImpl.kt` for those targets.

**Comment by hick209** [comment]
In case it's easy, while you are at it, would you mind setting things up for Android as well?

## Use case
In my company here there are people that are running clikt inside their test flavor of Android app, which they use to validate things in some e2e tests they run.

Because of this, we're currently stuck on clikt pre 4.0 and I'd love to bump to the latest.
Once I get some time I plan on trying to get the code for it into Mordant (and maybe clikt if necessary).

**Comment by hick209** [comment]
I've created an issue about this #165 with more details

**Comment by ajalt** [comment]
I added iOS in #168, watchOS and tvOS are blocked by https://github.com/JetBrains/markdown/pull/159

**Pull request closed without merging**